src/libs/actions/Task.js

```diff
@@ -186,28 +186,31 @@ function createTaskAndNavigate(currentUserEmail, currentUserAccountID, parentRep
 
 /**
  * Completes an open task
- * @param {string} taskReportID ReportID of the task
- * @param {string} taskTitle Title of the task
- * @param {string} originalLastMessageText Original last message text shown in LNH preview
+ * @param {Object} report
  */
-function completeTask(taskReportID, taskTitle, originalLastMessageText) {
-    const message = `completed task: ${taskTitle}`;
-    const completedTaskReportAction = ReportUtils.buildOptimisticTaskReportAction(taskReportID, CONST.REPORT.ACTIONS.TYPE.TASKCOMPLETED, message);
+function completeTask(report) {
+    const message = `completed task: ${ReportUtils.getReportName(report)}`;
+    const completedTaskReportAction = ReportUtils.buildOptimisticTaskReportAction(report.reportID, CONST.REPORT.ACTIONS.TYPE.TASKCOMPLETED, message);
 
     const optimisticData = [
         {
             onyxMethod: Onyx.METHOD.MERGE,
-            key: `${ONYXKEYS.COLLECTION.REPORT}${taskReportID}`,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${report.reportID}`,
             value: {
                 stateNum: CONST.REPORT.STATE_NUM.SUBMITTED,
                 statusNum: CONST.REPORT.STATUS.APPROVED,
                 lastMessageText: message,
+                lastMessageHtml: message,
```

Comment:
```suggestion
                lastMessageHtml: _.escape(message),
```
html and text are never the same. Here we should escape the text in case it contains html entities like `<` or `&`

Reply:
Not in this specific case right?  Since the `message` here is created on line `192` (similar to the 2 other methods) I think it doesn't contain any html so both 

I think that adding that `_.escape` has no cons so I'll test and add it : )

(same for the other 2 functions)

Reply:
The message contains a dynamic report name. If the report name is e.g. `Day & Night` then this would cause a problem because the html version of this text should be `Day &amp; Night` and not `Day & Night`.

Reply:
Yeah makes total sense, added that to all the `lastMessageHtml` I've added!